public_html/wp-content/plugins/camptix-attendance/addons/assets/attendance-ui.js

```diff
@@ -574,6 +587,76 @@ jQuery(document).ready(function($){
 			return false;
 		},
 
+		qr: function () {
+			this.$el.toggleClass('qr-scanner-active');
+			this.$menu.removeClass('dropdown');
+
+			var that = this;
+
+			let html5QrcodeScanner;
+			let lastScan;
+			let toggleView;
+
+			function onScanSuccess(decodedText, decodedResult) {
+				if (lastScan === decodedText) {
+					return;
+				}
+
+				lastScan = decodedText;
+
+				if (toggleView) toggleView.close();
+				let method = 'read';
+				let model = that.collection;
+				let options = {
+					data: { qrcode: decodedText },
+				};
+				let attendeeModel;
+
+				that.collection
+					.sync(method, model, options)
+					.done(
+						function (res) {
+							// not sure why this is an array
+							attendeeModel = new camptix.models.Attendee(res[0]);
+							toggleView = new camptix.views.AttendeeToggleView({
+								model: attendeeModel,
+								controller: that,
+							});
+							$(document.body).append(toggleView.render().el);
+						}.bind(that)
+					)
+					.fail(function (res) {
+						console.log('fail', res);
+					})
```

Comment:
Write some more helpful debug output or better remove it completely, when the PR is finished.